### PR TITLE
EVG-5887 Use an amboy job to prune the remote queue group

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
The first commit is a re-revert, so you only need to review the second.

File: environment.go [hunk 1]
```diff
@@ -338,7 +339,8 @@ func (e *envState) createGenerateTasksQueue(ctx context.Context) error {
 	e.remoteQueueGroup = remoteQueueGroup
 
 	e.closers["single-queue-generate-tasks"] = func(ctx context.Context) error {
-		e.generateTasksQueue.Runner().Close()
+		closerCtx := context.Background()
```
Comment (tychoish) [inline environment.go]: I don't think you want this to block forever. 

also change the name of the closer that you register

File: operations/service.go [hunk 2]
```diff
@@ -100,7 +100,9 @@ func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
 
 	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 15*time.Minute, time.Now(), opts, amboy.GroupQueueOperationFactory(
 		units.PopulateCatchupJobs(30),
-		units.PopulateHostAlertJobs(20)))
+		units.PopulateHostAlertJobs(20),
+		units.PopulatePruneRemoteQueueJobs(),
```
Comment (tychoish) [inline operations/service.go]: this seems like a bit infrequent?

File: units/crons.go [hunk 3]
```diff
@@ -764,3 +764,10 @@ func PopulateJasperCleanup(env evergreen.Environment) amboy.QueueOperation {
 		return queue.Put(NewJasperManagerCleanup(ts, env))
 	}
 }
+
+func PopulatePruneRemoteQueueJobs() amboy.QueueOperation {
+	return func(queue amboy.Queue) error {
+		ts := util.RoundPartOfHour(0).Format(tsFormat)
```
Comment (tychoish) [inline units/crons.go]: hourly?

File: units/prune_remote_group.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+func (j *pruneRemoteQueueGroup) Run(ctx context.Context) {
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+	if j.env == nil {
+		j.env = evergreen.GetEnvironment()
+	}
+	j.AddError(j.env.RemoteQueueGroup().Prune(ctx))
+	j.AddError(errors.Wrap(ctx.Err(), "context expired before prune finished"))
```
Comment (tychoish) [inline units/prune_remote_group.go]: shouldn't this be redundant?

File: environment.go (hunk 1 above)
Comment (bsamek) [reply environment.go]: done

File: operations/service.go (hunk 2 above)
Comment (bsamek) [reply operations/service.go]: the TTL is a week, so I don't think it needs to run very often

File: units/crons.go (hunk 3 above)
Comment (bsamek) [reply units/crons.go]: yeah, the TTL is quite long

File: units/prune_remote_group.go (hunk 4 above)
Comment (bsamek) [reply units/prune_remote_group.go]: it should be, but I think this is worth it for safety

File: environment.go [hunk 5]
```diff
@@ -338,8 +338,8 @@ func (e *envState) createGenerateTasksQueue(ctx context.Context) error {
 	}
 	e.remoteQueueGroup = remoteQueueGroup
 
-	e.closers["single-queue-generate-tasks"] = func(ctx context.Context) error {
-		closerCtx := context.Background()
+	e.closers["generate-tasks"] = func(ctx context.Context) error {
```
Comment (tychoish) [inline environment.go]: use the context from the function signature rather than making your own

Comment (bsamek) [reply environment.go]: done

**Comment by tychoish** [comment]
lets rebase. I think we should run prune more frequently, but the new queue implemenation. I'm fine with the code.

**Pull request closed without merging**